source/frontend/src/components/measurements/AreaContainer.tsx

```diff
@@ -0,0 +1,37 @@
+import { AreaForm } from './AreaForm';
+import AreaView from './AreaView';
+
+interface ViewOnlyProps {
+  landArea?: number;
+  unitCode?: string;
+
+  isEdditable?: never;
+  onChange?: never;
+}
+
+interface EditableProps {
+  landArea?: number;
+  unitCode?: string;
+
+  isEdditable: boolean;
+  onChange: (landArea: number, areaUnitTypeCode: string) => void;
+}
+
+type IAreaContainerProps = EditableProps | ViewOnlyProps;
+
+function isViewOnly(areaComponent: ViewOnlyProps | EditableProps): areaComponent is ViewOnlyProps {
```

Comment:
Is using a cast in the function definition necessary? (areaComponent is ViewOnlyProps).

Reply:
Good point. no need for the extra casting